keras_nlp/utils/tf_utils.py

```diff
@@ -67,3 +73,10 @@ def assert_tf_text_installed(symbol_name):
             f"{symbol_name} requires the `tensorflow-text` package. "
             "Please install with `pip install tensorflow-text`."
         )
+
+
+def is_tensor_type(x):
+    if pd is None:
+        return isinstance(x, (tf.Tensor, np.ndarray))
+    else:
+        return isinstance(x, (tf.Tensor, np.ndarray, pd.Series, pd.DataFrame))
```

Comment:
`pandas` is a tensor type? Or I guess it can be converted to one?

Reply:
convertible yes. this is copied from core Keras in terms of what types we handle as "tensor-like"

https://github.com/keras-team/keras/blob/2fd719d00ae7b26ed80b3345c5eb0264685e95aa/keras/engine/data_adapter.py#L1851-L1855

we had this for the pipeline utility already, this is just pulling it out to a standalone method so we can use it elsewhere.